packages/core/asyncComputed/index.ts

```diff
@@ -0,0 +1,57 @@
+import { ref, watchEffect, Ref } from 'vue-demi'
+
+/**
+ * Handle overlapping async evaluations
+ *
+ * @param cancelCallback The provided callback is invoked when a re-evaluation of the computed value is triggered before the previous one finished
+ */
+export type AsyncComputedOnCancel = (cancelCallback: () => void) => void
+
+/**
+ * A two-item tuple with the first item being a ref to the computed value and the second item holding a boolean ref, indicating whether the async computed value is currently (re-)evaluated
+ */
+export type AsyncComputedResult<T> = [Ref<T>, Ref<boolean>]
+
+/**
+ * Create an asynchronous computed dependency
+ *
+ * @param evaluationCallback     The promise-returning callback which generates the computed value
+ * @param defaultValue           A default value, used until the first evaluation finishes
+ */
+export function asyncComputed<T>(
+  evaluationCallback: (onCancel: AsyncComputedOnCancel) => T | Promise<T>,
+  defaultValue?: T,
+): AsyncComputedResult<T> {
+  let counter = 0
+  const current = ref(defaultValue) as Ref<T>
+  const evaluating = ref<boolean>(false)
+
+  watchEffect(async(onInvalidate) => {
+    counter++
+    const counterAtBeginning = counter
+    let hasFinished = false
+
+    try {
+      // Defer initial setting of `evaluating` ref
```

Comment:
Consider adding parameter `reset` (maybe different name) and adding:
```js
if (reset) {
  current.value = defaultValue
}
```
It would be useful for uses with `vue-router` to reset page to default loading state when page props change. Instead of page content staying the same until new data is loaded.

Reply:
I considered this use case, but decided against it because [boolean parameters are undesirable](https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/) and this case can dealt with pretty easily through an additional `computed`:

```js
function resettableAsyncComputed(evaluationCallback, defaultValue) {
  const [result, evaluating] = asyncComptued(evaluationCallback, defaultValue)
  return computed(() => evaluating.value ? defaultValue : result.value)
}
```

Reply:
Thanks for explanation. I really like this solution. This also allows for initial value and evaluating value to be different.

Reply:
Maybe this example could be added to docs? IMO this is pretty common use case.